### ISSUE TITLE
The "Harden the `chmod` function" rule doesn't match expected values

### ISSUE THREAD

**Issue by WhiteWinterWolf** [jvoisin/snuffleupagus#391]
The following rule from the default rules set does not manage to block expected commands for several reasons:

```
# Harden the `chmod` function
sp.disable_function.function("chmod").param("permissions").value_r("^[0-9]{2}[67]$").drop();
```

Tried using PHP 8, using the updated rules set with the parameter renamed from *mode* to *permission*.

## The regex is too strict

The regex as it is now would match neither "0777" (which is widely used by legitimate software) nor "07" (which would be a trivial way to by pass this filter).

The rule should only focus on the last byte, ie. worldwide permissions:

    sp.disable_function.function("chmod").param("permissions").value_r("[67]$").drop();

However this still doesn't work.

## The regex is applied to the decimal value

As the title says, the regex is applied to the decimal value received by the function instead of the octal one written in the source code.

For instance, the decimal value for "0777" is "511". Updating the regex this ways successfully catches a `chmod("foo", 0777);`:

    sp.disable_function.function("chmod").param("permissions").value_r("[1]$").drop();

**Comment by jvoisin** [comment]
Can you please send a pull-request to update the default rules?

**Comment by WhiteWinterWolf** [comment]
I wish I had a correct rule to provide ;) !

Is there any way to tell Snuffleupagus to apply the regex on the octal value instead of the decimal one? Or a way to use bitwise operators in the match rule (something equivalent to `& 3`)?

**Comment by jvoisin** [comment]
Unfortunately not, but I guess that matching on `[137]` should do the job?

**Comment by WhiteWinterWolf** [comment]
No, you cannot directly deduce the value of the two rightmost bits simply by looking at the last digit of a decimal value. For instance, the octal value 0666 is 438 in decimal. If you take less common mode value, 016 is 14, 017 is 15, etc.

The only "solution" I can think of would be to target an block specifically 0777 and 0666 (resp. 511 and 438) exact values using two simple `value()` match rules, as misbehaving legitimate applications usually use those rights. This won't prevent any malicious permission changes (for data exfiltration or to allow code execution using the write or execute permission), but it should still help against the most common scenario.

**Comment by jvoisin** [comment]
This has been ~fixed via #392!
